### PR TITLE
Add a secret_headers input in uri module

### LINKED ISSUE #73244: Add a header_secret input for uri module
##### SUMMARY
We use a lot uri module, and we send API token in the header. These secrets need to be hidden on the log, so we are forced to mark the entire task as no_log: true missing many logs.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
uri module

##### ADDITIONAL INFORMATION
Just add a header_secret input in uri with no_log=True optional argument.
It would be merged with the header input.

### PR DESCRIPTION
**Pull request by PierreLesouhaitier**
##### SUMMARY
Fixes #73244

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Update the uri module

### CODE REVIEW

File: lib/ansible/modules/uri.py [hunk 1]
```diff
@@ -121,6 +121,11 @@
           generated by supplying C(json) or C(form-urlencoded) for I(body_format).
     type: dict
     version_added: '2.1'
+  headers_scret:
```
Comment (Shrews) [inline lib/ansible/modules/uri.py:124]: misspelled here

File: lib/ansible/modules/uri.py [hunk 2]
```diff
@@ -121,6 +121,11 @@
...
+    description:
+        - Add secret in HTTP headers.
+          Anything added in headers_secret will get in the headers but won't be displayed in the logs.
```
Comment (Shrews) [inline lib/ansible/modules/uri.py:127]: need a `version_added`

**Comment by Shrews** [comment]
We will need a changelog and tests before we consider merging this. We may also want to bikeshed on the `headers_secret` name.

File: lib/ansible/modules/uri.py (hunk 2 above)
Comment (Akasurde) [inline lib/ansible/modules/uri.py:127]:
```suggestion
        - Anything added in I(headers_secret) will get in the headers but will not be displayed in the logs.
```

File: lib/ansible/modules/uri.py (hunk 1 above)
Comment (Akasurde) [inline lib/ansible/modules/uri.py:124]: Add a changelog entry [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) and integration test [here](https://github.com/ansible/ansible/tree/devel/test/integration/targets/uri).

**Comment by Akasurde** [comment]
@PierreLesouhaitier Are you still working on this? Thanks.

needs_info

**Pull request closed without merging**